doc/source/development_and_release_processes.rst

```diff
@@ -0,0 +1,203 @@
+packages* are released:
+
+#. Create ``release`` branches for *auto-generated packages* and publish as release candidates to public PyPI.
+#. Create Pull Requests in *idiomatic package* repositories to update ``main`` to depend on the *auto-generated
+   package* release candidate releases. Ensure CI passes for *idiomatic packages*.
+#. Publish *auto-generated packages* as stable releases to public PyPI.
+#. Create Pull Requests in *idiomatic package* repositories to update ``main`` to depend on the *auto-generated
+   package* stable releases. Ensure CI passes for *idiomatic packages*.
+#. Create ``release`` branches for *idiomatic packages* and publish as release candidates to public PyPI.
+
+   .. note::
+      At this point release branches should be created for *idiomatic packages*. However, during hardening these release
+      branches still run CI against the *development test VM*.
+
+   .. note::
+      *Idiomatic package* release candidates must depend on stable release versions of *auto-generated packages* only.
+      They must not depend on pre-release versions of *auto-generated packages*.
+
+   .. note::
+      When performing pre-releases of packages with towncrier, the Ansys action creates a Pull Request to update
+      CHANGELOG.md on ``main``, and commits the CHANGELOG.md changes directly to the release branch. The commit to the
+      release branch should be reverted, and the Pull Request to merge the changes to ``main`` should be closed and
+      not merged.
+
+The stable *auto-generated packages* and release candidate *idiomatic packages* can now be used for the final release of
+internal packages. It is recommended to make any closed-source *idiomatic package* release candidate dependencies
+flexible enough such that they match subsequent stable releases of those *idiomatic packages*.
+
+Checklist for completing this phase:
+
+* |chkbx| The *development test VM* is running an up-to-date development version of Granta MI.
+* |chkbx| CI for all *idiomatic packages* is passing.
+* |chkbx| *Idiomatic package* documentation has been reviewed to update changes in dependencies, for example Python
+  version support, minimum Granta MI version.
+* |chkbx| ``release`` branches have been created for *auto-generated packages* and *idiomatic packages*.
+* |chkbx| Stable *auto-generated packages* are available on PyPI.
+* |chkbx| Internal packages with runtime dependencies on *auto-generated packages* depend on the stable PyPI releases.
+* |chkbx| *Idiomatic packages* depend on stable releases of *auto-generated packages* only.
+* |chkbx| Release candidate *idiomatic packages* are available on PyPI.
+* |chkbx| Internal packages with runtime dependencies on *idiomatic packages* depend on the release candidate PyPI
+  releases.
+
+Phase 3: Pre-release
+~~~~~~~~~~~~~~~~~~~~
+
+#. Update the *production test VM* to run the Granta MI release validated during the hardening phase.
```

Comment:
> https://github.com/ansys/pygranta/issues/39#issuecomment-2371424921: Some packages include links to the Ansys help site and include a version. The release checklist for such packages should mention updating URLs to the latest supported MI version.

We should add something in here to say that in Phase 3 we check all docs links and update them if relevant.